### DISCUSSION TITLE
[ch 05. 트랜잭션과 잠금] 트랜잭션 내 외부 통신 작업 제거

### DISCUSSION THREAD

**Discussion by dojinyou** [Learning-Is-Vital-In-Development/23-6-RealMySQL#20] [Q&A]
p.159 트랜잭션 주의사항에서 "네트워크를 통해 원격 서버와 통신하는 등 외부와의 통신 작업은 트랜잭션 내에서 제거해야 한다. "는 말이 있는데, AWS S3 파일 업로드 등의 작업과 같은 클라우드 환경에서의 작업 또한 이에 해당하는지 궁금합니다.

**Comment by tlarbals824** [comment]
외부와의 통신에서 시간이 길어지면 트랜잭션 시간이 길어지고 트랜잭션 시간이 길어지면 좋지 않기 때문에 분리해야한다고 생각합니다.

**Reply by dojinyou** [reply]
여기서 "외부"라고 하는 것은 해당 로직이 실행되고 있는 서버 및 연결된 서버(DB, Redis 등)을 제외한 나머지를 지칭한다고 생각함!
